### PR TITLE
Pbi 35 42 updateinformation on existing inventory items

### PR DESCRIPTION
**Pull request by bryannaufal**
## Describe your changes
Create test for update product accessor
Create test for update price accessor
Create test for update product service
Create test for update price service
Create implementation for update product accessor
Create implementation for update price accessor
Create implementation for update product service
Create implementation for update price service

## Issue ticket number and link
ADU-35: [Update information on existing inventory items API](https://linear.app/adudu/issue/ADU-35/update-information-on-existing-inventory-items-api)
ADU-42: [UT for update information on existing inventory items API](https://linear.app/adudu/issue/ADU-42/ut-for-update-information-on-existing-inventory-items-api)

### CODE REVIEW

File: internal/product/accessor.go [hunk 1]
```diff
@@ -103,28 +105,188 @@ func (p *postgresProductAccessor) GetProductsByVendor(
...
+}
+
+
 // newPostgresProductAccessor is only accessible by the Product package
 // entrypoint for other verticals should refer to the interface declared on service
-func newPostgresProductAccessor(db database.DBConnector) *postgresProductAccessor {
+func newPostgresProductAccessor(db database.DBConnector, clock clock.Clock) *postgresProductAccessor {
 	return &postgresProductAccessor{
 		db: db,
+        clock: clock,
```
Comment (valeelim) [inline internal/product/accessor.go:290]: tolong di indent ngab kowkowokw

File: internal/product/service.go [hunk 2]
```diff
@@ -4,10 +4,14 @@ package product
...
 	"context"
 	"kg/procurement/internal/common/database"
+
+	"github.com/benbjohnson/clock"
 )
 
 type productDBAccessor interface {
 	GetProductsByVendor(ctx context.Context, vendorID string, spec GetProductsByVendorSpec) ([]Product, error)
+	UpdatePrice(ctx context.Context, price Price) (*Price, error)
+	UpdateProduct(ctx context.Context, payload Product) (*Product, error)
```
Comment (valeelim) [inline internal/product/service.go:14]: Return object-nya langsung aja
```suggestion
	UpdatePrice(ctx context.Context, price Price) (Price, error)
	UpdateProduct(ctx context.Context, payload Product) (Product, error)
```

File: router/product.go [hunk 3]
```diff
@@ -39,4 +39,75 @@ func NewProductEngine(
 
 		ctx.JSON(http.StatusOK, res)
 	})
+
+	r.PUT(cfg.UpdateProduct, func(ctx *gin.Context) {
+		id := ctx.Request.PathValue("id")
+
+		spec := &product.PutProductSpec{}
```
Comment (valeelim) [inline router/product.go:46]: ini ga perlu & (?)

File: router/product.go [hunk 4]
```diff
@@ -39,4 +39,75 @@ func NewProductEngine(
...
+			ctx.JSON(http.StatusInternalServerError, gin.H{
+				"error": err.Error(),
+			})
+		}
+		ctx.JSON(http.StatusOK, res)
+	})
+
+	r.PUT(cfg.UpdatePrice, func(ctx *gin.Context) {
+		id := ctx.Request.PathValue("id")
+		spec := &product.PutPriceSpec{}
```
Comment (valeelim) [inline router/product.go:72]: yg ini juga (?)

File: cmd/config/config.go [hunk 5]
```diff
@@ -35,6 +35,8 @@ type VendorRoutes struct {
 
 type ProductRoutes struct {
 	GetProductsByVendor string `mapstructure:"get-products-by-vendor" validate:"required"`
+	UpdateProduct string `mapstructure:"update-product" validate:"required"`
+	UpdatePrice string `mapstructure:"update-price" validate:"required"`
```
Comment (valeelim) [inline cmd/config/config.go:39]: `config.example`-nya kykny lupa di push jg ya?

File: router/product.go (hunk 3 above)
Comment (ferryhieronimus) [reply router/product.go]: Kapan sebaiknya return pointer dan return value?

File: cmd/config/config.go (hunk 5 above)
Comment (bryannaufal) [reply cmd/config/config.go:39]: Maaf terlewat, saat ini sudah ditambahkan

**Pull request merged**